mcda/mcda_run.py

```diff
@@ -1,377 +1,164 @@
 #! /usr/bin/env python3
 
-import logging
+"""
+This script serves as the main entry point for running all pieces of functionality in a consequential way by
+following the settings given in the configuration file 'configuration.json'.
+
+Usage (from root directory):
+    $ python3 -m mcda.mcda_run -c configuration.json
+"""
+
 import time
-import sys
-import os
+import logging
 
 from mcda.configuration.config import Config
-from mcda.utils import *
-from mcda.utils_for_parallelization import *
-from mcda.mcda_without_robustness import MCDAWithoutRobustness
-from mcda.mcda_with_robustness import MCDAWithRobustness
+from mcda.utils.utils_for_main import *
+from mcda.utils.utils_for_plotting import *
+from mcda.utils.utils_for_parallelization import *
+
+log = logging.getLogger(__name__)
 
-formatter = '%(levelname)s: %(asctime)s - %(name)s - %(message)s'
-logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, format=formatter)
+FORMATTER: str = '%(levelname)s: %(asctime)s - %(name)s - %(message)s'
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, format=FORMATTER)
 logger = logging.getLogger("ProMCDA")
 
-class UserStoppedInfo(Exception):
-    pass
+# noinspection PyTypeChecker
+def main(input_config: dict):
+    """
+        Execute the ProMCDA (Probabilistic Multi-Criteria Decision Analysis) process.
+
+        Parameters:
+        - input_config (dictionary): Configuration parameters for the ProMCDA process.
 
-def main(input_config: dict, user_input_callback=input):
+        Raises:
+        - ValueError: If there are issues with the input matrix, weights, or indicators.
+
+        This function performs the ProMCDA process based on the provided configuration.
+        It handles various aspects such as the sensitivity analysis and the robustness analysis.
+        The results are saved in output files, and plots are generated to visualize the scores and rankings.
+
+        Note: Ensure that the input matrix, weights, polarities and indicators (with or without uncertainty)
+        are correctly specified in the input configuration.
+        """
 
     is_sensitivity = None
     is_robustness = None
     is_robustness_indicators = 0
     is_robustness_weights = 0
-    norm_random_weights = []
     marginal_pdf = []
     num_unique = []
-    iterative_random_w_means = {}
-    iterative_random_w_stds = {}
-    iterative_random_w_means_normalized = {}
-    iterative_random_w_stds_normalized = {}
 
+    t = time.time()
 
+    # Extracting relevant configuration values
     config = Config(input_config)
     input_matrix = read_matrix(config.input_matrix_path)
     polar = config.polarity_for_each_indicator
     is_sensitivity = config.sensitivity['sensitivity_on']
     is_robustness = config.robustness['robustness_on']
     mc_runs = config.monte_carlo_sampling["monte_carlo_runs"]
+
+    # Check for sensitivity-related configuration errors
     if is_sensitivity == "no":
```

Comment:
just a small thing: I would prefer using boolean primitive values (true/false), is that possible in the configuration?

Reply:
No because is a JSON. The problem is only temporary though, because we are moving to a library (with no more configuration files).